lib/licenses.nix

```diff
@@ -899,8 +870,178 @@ in mkLicense lset) ({
     spdxId = "ZPL-2.1";
     fullName = "Zope Public License 2.1";
   };
+
 } // {
-  # TODO: remove legacy aliases
+  /*
+    Below is the list of known license exceptions introduced with SPDX 2.0
+    https://spdx.org/licenses/exceptions-index.html
+  */
+
+  autoconf-exception2 = {
+    spdxId = "Autoconf-exception-2.0"; # Typically used with GPL-2.0
+    fullName = "Autoconf exception 2.0";
+  };
+
+  autoconf-exception3 = {
+    spdxId = "Autoconf-exception-3.0"; # Typically used with GPL-3.0
+    fullName = "Autoconf exception 3.0";
+  };
+
+  bison-exception = {
+    spdxId = "Bison-exception-2.2"; # Typically used with GPL-2.0 or GPL-3.0
+    fullName = "Bison exception 2.2";
+  };
+
+  classpath-exception = {
+    spdxId = "Classpath-exception-2.0"; # Applies to GPL-2.0 / GPL-3.0
+    fullName = "Classpath exception 2.0";
+  };
+
+  clisp-exception = {
+    spdxId = "CLISP-exception-2.0"; # Typically used with GPL-2.0
+    fullName = "CLISP exception 2.0";
+  };
+
+  fltk-exception = {
+    spdxId = "FLTK-exception"; # Applies to LGPL-2.0-only
+    fullName = "FLTK exception";
+  };
+
+  font-exception = {
+    spdxId = "Font-exception-2.0"; # Typically used with GPL-2.0-or-later
+    fullName = "Font exception 2.0";
+  };
+
+  gcc-exception2 = {
+    spdxId = "GCC-exception-2.0"; # Typically used with GPL-2.0-or-later
+    fullName = "GCC Runtime Library exception 2.0";
+  };
+
+  gcc-exception31 = {
+    spdxId = "GCC-exception-3.1"; # Typically used with GPL-3.0
+    fullName = "GCC Runtime Library exception 3.1";
+  };
+
+  gpl3-linking-exception = {
+    spdxId = "GPL-3.0-linking-exception"; # Applies to GPL-3.0
+    fullName = "GPL-3.0 Linking Exception";
+  };
+
+  gpl3-linking-source-exception = {
```

Comment:
I'm not sure if we want to go with dashes or lowerCamelCase.
We already had `llvm-exception` in licenses.nix and I followed that, but the licenses themselves mostly use camel case.
I can rework this PR if need be.

For longer names like this one it would become `gpl3LinkingSourceException`.
Not sure that's more readable than `gpl3-linking-source-exception`.